src/main.rs

```diff
@@ -64,16 +63,18 @@ fn main() {
       .lint(file_name, source_code, rules)
       .expect("Failed to lint");
 
-    diagnostics.extend(file_diagnostics)
-  }
-
-  if !diagnostics.is_empty() {
-    for d in diagnostics.iter() {
-      eprintln!(
-        "error: {} ({}) at {}:{}:{}",
-        d.message, d.code, d.location.filename, d.location.line, d.location.col
-      );
+    error_counts += file_diagnostics.len();
+    if !file_diagnostics.is_empty() {
+      eprintln!("{} =>", file_diagnostics[0].location.filename);
+      for d in file_diagnostics.iter() {
+        eprintln!("  {}| {}", d.location.line, d.line_src);
+        eprintln!(
+          "  {}^",
+          " ".repeat(d.location.line.to_string().len() + d.location.col + 2)
+        );
+        eprintln!("  ({}) {}", d.code, d.message);
+      }
```

Comment:
Cool :+1: can you keep the old formatting as was and additionally display code snippet on the next line:

Eg:
```
// now:
▶ target/debug/dlint ../deno/std/http/server.ts
../deno/std/http/server.ts =>
  93|       } catch {}
                    ^
  (noEmpty) Empty block statement
  111|     while ((await body.read(buf)) !== null) {}
                                                   ^
  (noEmpty) Empty block statement
  120|   constructor(public listener: Listener) {}
                                                ^
  (noEmpty) Empty block statement
Found 3 problems
```

```
// new:
▶ target/debug/dlint ../deno/std/http/server.ts
error: Empty block statement (noEmpty) at ../deno/std/http/server.ts:93:0
93|       } catch {}
                    ^
...
Found 3 problems
```

I'll be refactoring AST parser soon so I'll figure a way to obtain code snippet plus surrounding lines

Reply:
Ah I see that you show all errors from a single file collapsed together. That's probably better approach